### ISSUE TITLE
quagga cannot start on image installed using sonic2sonic upgrade

### ISSUE THREAD

**Issue by marian-pritsak** [Azure/sonic-buildimage#579]
```
SONiC Software Version: SONiC.HEAD.240-8f34839
Distribution: Debian 8.8
Kernel: 3.16.0-4-amd64
Build commit: 8f34839
Build date: Tue May  9 07:45:48 UTC 2017
Built by: johnar@jenkins-worker-1

Docker images:
REPOSITORY                TAG                 IMAGE ID            SIZE
docker-orchagent-mlnx     latest              8f2dc4757732        256 MB
docker-syncd-mlnx         latest              1907b0ad3a57        405.7 MB
docker-dhcp-relay         latest              a84b35132b9e        251.6 MB
docker-database           latest              5ae54585019b        219.5 MB
docker-snmp-sv2           latest              a870210fac95        289.3 MB
docker-teamd              latest              5ed3986424b4        253.5 MB
docker-platform-monitor   latest              bc3f2854284f        268.6 MB
docker-lldp-sv2           latest              811ef1a27405        255.1 MB
docker-fpm-quagga         latest              aaef6cc119d2        260 MB
```

Daemons zebra and bgpd are not running
```
root@arc-switch1026:/home/admin# docker exec -it bgp ps ax
  PID TTY      STAT   TIME COMMAND
    1 ?        Ss+    0:00 /usr/bin/python /usr/bin/supervisord
   25 ?        Sl     0:00 /usr/sbin/rsyslogd -n
   51 ?        Ss     0:00 /usr/lib/quagga/watchquagga --daemon zebra bgpd
   53 ?        Sl     0:00 fpmsyncd
   70 ?        Rs+    0:00 ps ax
```

```
# docker logs bgp                                                                                                                                                                                                                                
/usr/lib/python2.7/dist-packages/supervisor/options.py:296: UserWarning: Supervisord is running as root and it is searching for its configuration file in default locations (including its current working directory); you probably want to specify a "-c" argument specifying an absolute path to a configuration file for improved security.
  'Supervisord is running as root and it is searching '
2017-05-09 12:18:00,785 CRIT Supervisor running as root (no user in config file)
2017-05-09 12:18:00,786 WARN Included extra file "/etc/supervisor/conf.d/supervisord.conf" during parsing
2017-05-09 12:18:00,848 INFO RPC interface 'supervisor' initialized
2017-05-09 12:18:00,850 CRIT Server 'unix_http_server' running without any HTTP authentication checking
2017-05-09 12:18:00,852 INFO supervisord started with pid 1
2017-05-09 12:18:01,855 INFO spawned: 'start.sh' with pid 8
2017-05-09 12:18:02,858 INFO success: start.sh entered RUNNING state, process has stayed up for > than 1 seconds (startsecs)
2017-05-09 12:18:04,390 INFO spawned: 'rsyslogd' with pid 25
2017-05-09 12:18:05,393 INFO success: rsyslogd entered RUNNING state, process has stayed up for > than 1 seconds (startsecs)
2017-05-09 12:18:05,455 CRIT reaped unknown pid 42)
2017-05-09 12:18:05,504 CRIT reaped unknown pid 46)
2017-05-09 12:18:05,641 INFO spawned: 'fpmsyncd' with pid 53
2017-05-09 12:18:06,667 INFO success: fpmsyncd entered RUNNING state, process has stayed up for > than 1 seconds (startsecs)
2017-05-09 12:18:06,681 INFO exited: start.sh (exit status 0; expected)
```

**Comment by marian-pritsak** [comment]
In case SAME image is installed from ONIE, everything works fine

**Comment by marian-pritsak** [comment]
@jleveque did you try to reproduce this?

**Comment by jleveque** [comment]
I have reproduced the issue. Zebra and bgpd fail to start because the ownership of the `/var/run/quagga` directory of images installed using the SONiC-to-SONiC update is incorrect. It should be owned by the `quagga` user, but it is owned by a different user. This prevents the processes from writing their PID files to `/var/run/quagga`, so the processes exit in error. I have not determined a reason for the incorrect ownership yet.

**Comment by lguohan** [comment]
I think there is some problem in the docker.tar.gz file.

```
./aufs/diff/c3948b1fd959e3b937d497769cd18a8f60f8ce1d2ddd6893dba908358ad11e2e/run/quagga/^@^@^@^@^@^@^@^@^@
^@^@^@0000755^@0000152^@0000157^@00000000000^@13111135400^@024446^@ 5^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@
^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^
@^@^@^@^@^@^@^@^@^@^@^@^@^@ustar  ^@sshd^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@ssh^@^@^@^@^@^@^@^@^@^@^@
^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^
@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@
^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^
@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@.
```

the ```uid``` and ```uid``` are correct, but not the ```uname``` and ```gname```. 

according to the standard,

https://www.gnu.org/software/tar/manual/html_node/Standard.html

The magic field indicates that this archive was output in the P1003 archive format. If this field contains TMAGIC, the uname and gname fields will contain the ASCII representation of the owner and group of the file respectively. If found, the user and group IDs are used rather than the values in the uid and gid fields.

**Comment by lguohan** [comment]
addressed in #626
